r/vignettes/developers/install_details.Rmd

```diff
@@ -0,0 +1,90 @@
+# How the R package is installed
+
+In order for the `arrow` R package to work, it needs the Arrow C++ library, 
+also known as libarrow.  There are a number of scripts that are triggered 
+when `R CMD INSTALL .` is run and for Arrow users, these should all just work 
+without configuration and pull in the most complete pieces (e.g. official 
+binaries that we host).
+
+An overview of these scripts is shown below:
+
+* `configure` and `configure.win` - these scripts are triggered during
+`R CMD INSTALL .` on non-Windows and Windows platforms, respectively. They
+handle finding the libarrow, setting up the build variables necessary, and
+writing the package Makevars file that is used to compile the C++ code in the R
+package.
+
+* `tools/nixlibs.R` - this script is sometimes called by `configure` on Linux
```

Comment:
Would it be helpful to make a mention of the conditions where it is called? (e.g. if libarrow is not already found on the system, if libarrow is not being downloaded as a binary, etc.)?

Reply:
Done

Reply:
We could also mention `tools/winlibs.R`, which is the Windows equivalent. This will either find or download binaries of libarrow; not set up any build process.

Reply:
Good point - in splitting this content out from the "Installing on Linux" vignette I'd made it sounds more generic, so it's worth adding that stuff back in, will do!

Reply:
@wjones127 Have added the section on the Windows equivalent - does it (and the diagram) look right to you?

Reply:
Diagram looks good. Had a correction on the text though. Thanks for including this.